build.sbt

```diff
@@ -553,7 +553,7 @@ lazy val dynamic = projectMatrix
     Compile / smithySpecs := Seq(
       (ThisBuild / baseDirectory).value / "modules" / "dynamic" / "smithy" / "dynamic.smithy"
     ),
-    Compile / sourceGenerators := Seq(genSmithyScala(Compile).taskValue),
+    genSmithy(Compile),
```

Comment:
I had a theory on why it broke down, but it doesn't appear to be true. I thought maybe having `resourceGenerators` here and in `core` (`dynamic` dependency) would put conflicting files on the class path and only one would be picked up

but it does not seem to be the case

Reply:
I sort of worked around it by not applying it across the board, but I'm unsatisfied

we figured out the issue with `http4s`: the test loads a file directly, and, after the change, the file is picked up by the discoverModels because it's copied in the resources

but for the `dynamic` issue, I still can't figure out: you add 3 files to the class path:
```
dynamic.smithy
manifest
smithy4s.tracking.smithy
```

And then suddenly, `alloy#discriminated` is unresolvable